PayloadPkg/OsLoader/OsLoader.c

```diff
@@ -11,80 +11,68 @@ UINT32  *mPreOsCheckerEntry;
+  IN      UINT32         ImageType
   )
 {
-  IAS_HEADER                 *IasImage;
-  UINT32                     NumFiles;
-  IMAGE_DATA                 File[MAX_IAS_SUB_IMAGE];
+  EFI_STATUS                 Status;
   UINTN                      InitrdSize;
   VOID                       *InitrdData;
-  UINT32                     ImageType;
   UINT16                     Index;
   UINT16                     ModuleIndex;
   LINUX_IMAGE                *LinuxImage;
   MULTIBOOT_IMAGE            *MultiBoot;
   COMMON_IMAGE               *CommonImage;
   PLATFORM_SERVICE           *PlatformService;
-  EFI_STATUS                 Status;
 
   PlatformService = NULL;
 
-  IasImage = IsIasImageValid (LoadedImage->IasImage.Addr, LoadedImage->IasImage.Size, LoadedImage->ImageHash);
-  if (IasImage == NULL) {
-    DEBUG ((DEBUG_INFO, "Image given is not a valid IAS image\n"));
-    return EFI_LOAD_ERROR;
-  }
-  AddMeasurePoint (0x4080);
-
-  ZeroMem (File, sizeof (File));
-  NumFiles = IasGetFiles (IasImage, sizeof (File) / sizeof ((File)[0]), File);
-  DEBUG ((DEBUG_INFO, "IAS size = 0x%x, file number: %d\n", LoadedImage->IasImage.Size, NumFiles));
-
   if (NumFiles == 1) {
-    // This is not valid IAS use case, at least 2 files in IAS image.
+    // This is not valid IAS/PODS use case, at least 2 files in image.
     // Support this only for test
     CommonImage                = &LoadedImage->Image.Common;
     CommonImage->BootFile.Addr = File[0].Addr;
     CommonImage->BootFile.Size = File[0].Size;
     if (IsMultiboot (File[0].Addr)) {
       LoadedImage->Flags |= LOADED_IMAGE_MULTIBOOT;
-      DEBUG ((DEBUG_INFO, "One multiboot file in IAS file .... \n"));
+      DEBUG ((DEBUG_INFO, "One multiboot file in IAS/PODS file .... \n"));
     } else if (IsTePe32Image (File[0].Addr, NULL) && \
                (* (UINT32 *)File[0].Addr == EFI_IMAGE_DOS_SIGNATURE)) {
       // Add extra check to ensure it is a PE32 image generated from payload build.
       // Please note vmlinuxz is also following PE32 format, but it should
       // be handled as Linux image boot path
       LoadedImage->Flags |= LOADED_IMAGE_PE32;
-      DEBUG ((DEBUG_INFO, "One PE32 file in IAS file .... \n"));
+      DEBUG ((DEBUG_INFO, "One PE32 file in IAS/PODS file .... \n"));
     } else if (IsValidFvHeader ((VOID *)File[0].Addr))  {
       LoadedImage->Flags |= LOADED_IMAGE_FV;
-      DEBUG ((DEBUG_INFO, "One FV file in IAS file .... \n"));
+      DEBUG ((DEBUG_INFO, "One FV file in IAS/PODS file .... \n"));
     } else {
       LoadedImage->Flags |= LOADED_IMAGE_LINUX;
-      DEBUG ((DEBUG_INFO, "One file in IAS file, take it as bzImage .... \n"));
+      DEBUG ((DEBUG_INFO, "One file in IAS/PODS file, take it as bzImage .... \n"));
     }
     return EFI_SUCCESS;
   }
 
-  ImageType = IAS_IMAGE_TYPE (IasImage->ImageType);
-  DEBUG ((DEBUG_INFO, "IAS Image Type = 0x%x\n", ImageType));
-  if (ImageType == IAS_TYPE_CLASSIC) {
+  if ((ImageType & 0xF) == IAS_TYPE_CLASSIC) {
```

Comment:
can we pass in the processed ImageType to this function ?
It means the caller do the ImageType conversion before calling it.